### PR TITLE
Check of Subscription ID before sending request to mongo layer

### PR DESCRIPTION
**Pull request by kzangeli**
Fixes issue #1552 

Checking subscription ID before letting it pass down to mongo layer, for GET "/v2/subscriptions/XXX"

### CODE REVIEW

File: src/lib/mongoBackend/mongoGetSubscriptions.cpp [hunk 1]
```diff
@@ -260,14 +262,20 @@ void mongoGetSubscription
   const std::string&                  tenant
 )
 {
-  bool  reqSemTaken = false;
+  bool         reqSemTaken = false;
+  std::string  err;
+
+  if ((err = idCheck(idSub)) != "OK")
```
Comment (fgalan) [inline src/lib/mongoBackend/mongoGetSubscriptions.cpp]: I would suggest to use safeGetSubId() in the checking at mongoBackend layer due to two reasons:

1. Homogeneity (similar checkings in other parts of mongoBackend are using that method).
2. Using a different method to the one used at parsing stage. This kind of "HA" will mean that is some fail occur in the checking done by idCheck(), then safeGetSubId() could catch (or the opossite).

File: CHANGES_NEXT_RELEASE [hunk 2]
```diff
@@ -22,3 +22,4 @@
 - Fix: no longer adding subscriptions from all tenants in the cache if broker isn't multitenant (Issue #1555)
 - Fix: resetting temporal counters at synching the subscription cache (Issue #1562)
 - Fix: using the text "too many sbuscriptions" at cache statistics operations in the case of too many results
+- Fix: crash due to subscription ID not conforming to supposed syntax (Issue #1552)
```
Comment (fgalan) [inline CHANGES_NEXT_RELEASE]: - Fix: crash due to subscription ID not conforming to supposed syntax in GET /v2/subscriptions/<id> operation (Issue #1552)

The "in GET /v2/subscriptions/<id> operation" could be relevant given that, as far as I understand, this problem doesn't happen in similar operations of the NGSIv1 family.

Comment (kzangeli) [inline CHANGES_NEXT_RELEASE]: a8c27f2

File: src/lib/mongoBackend/mongoGetSubscriptions.cpp (hunk 1 above)
Comment (kzangeli) [inline src/lib/mongoBackend/mongoGetSubscriptions.cpp]: Fixed in a8c27f2

**Comment by fgalan** [comment]
LGTM

**Pull request merged**